Add API test coverage for 283722@main

pxlcoder opened the pull request:
#### a10ee6a9b1ee5949ced6cfdab1a393243e65a9b3
<pre>
Add API test coverage for 283722@main
<a href="https://bugs.webkit.org/show_bug.cgi?id=281935">https://bugs.webkit.org/show_bug.cgi?id=281935</a>
<a href="https://rdar.apple.com/138440875">rdar://138440875</a>

Reviewed by Abrar Rahman Protyasha and Richard Robinson.

283722@main landed without any test coverage. Add an API test to help avoid
future regressions.

Verified that the test passes on main, and fails (times out) when reverting
283722@main and 283725@main (which also fixed the same issue in a different way).

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm:
(TEST(WritingTools, CompositionWithRestart)):

Web content crashes were encountered when performing two successive rewrites.

Add a test that makes Writing Tools API calls to match the system behavior when
performing two successive rewrites. Allow time for the animations to run to
completion.

Canonical link: <a href="https://commits.webkit.org/285590@main">https://commits.webkit.org/285590@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/979f109c59fdbb80fecfcc7842af9948e7968d6b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/73152 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/52581 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/25960 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/77373 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/24390 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/75267 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/60386 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/363 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/57480 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/15962 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/76219 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/47510 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/62948 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/37897 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/44149 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/20423 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/22719 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/66002 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/20777 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/79034 "Built successfully") | 
| | [⏳ 🛠 vision ](https://ews-build.webkit.org/#/builders/visionOS-2-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/49 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/65914 "Passed tests") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/608 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/62954 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/65196 "Passed tests") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/7193 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11270 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/431 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/460 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/445 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/462 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm:
```diff
@@ -1016,6 +1016,44 @@ static void checkColor(WebCore::CocoaColor *color, std::optional<ColorExpectatio
+
+    __block bool finished = false;
+
+    [[webView writingToolsDelegate] willBeginWritingToolsSession:session.get() requestContexts:^(NSArray<WTContext *> *contexts) {
```
super minor, but I've been starting to get into the pattern of putting as little stuff into the completion handlers as possible (see `IntelligenceTextEffectCoordinatorDelegate_TextPreviewsForRange`) since I think it makes it slightly more readable, but doesn't really matter

pxlcoder replied:
Will keep this in mind moving forward, but I'm going to land this as-is for now.

pxlcoder committed:
Add API test coverage for 283722@main
https://bugs.webkit.org/show_bug.cgi?id=281935
rdar://138440875

Reviewed by Abrar Rahman Protyasha and Richard Robinson.

283722@main landed without any test coverage. Add an API test to help avoid
future regressions.

Verified that the test passes on main, and fails (times out) when reverting
283722@main and 283725@main (which also fixed the same issue in a different way).

* Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm:
(TEST(WritingTools, CompositionWithRestart)):

Web content crashes were encountered when performing two successive rewrites.

Add a test that makes Writing Tools API calls to match the system behavior when
performing two successive rewrites. Allow time for the animations to run to
completion.

Canonical link: https://commits.webkit.org/285590@main
